ui/apps/platform/src/Containers/VulnMgmt/Reports/Form/CollectionSelection.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { useMemo, useState, ReactElement, useCallback } from 'react';
+import {
+    Button,
+    ButtonVariant,
+    Flex,
+    FlexItem,
+    Select,
+    SelectOption,
+    SelectProps,
+    SelectVariant,
+} from '@patternfly/react-core';
+import sortBy from 'lodash/sortBy';
+import uniqBy from 'lodash/uniqBy';
+
+import FormLabelGroup from 'Components/PatternFly/FormLabelGroup';
+import { Collection, listCollections } from 'services/CollectionsService';
+import CollectionsFormModal from 'Containers/Collections/CollectionFormModal';
+import { useCollectionFormSubmission } from 'Containers/Collections/hooks/useCollectionFormSubmission';
+import { useFormik } from 'formik';
+import useSelectToggle from 'hooks/patternfly/useSelectToggle';
+import { usePaginatedQuery } from 'hooks/usePaginatedQuery';
+
+const COLLECTION_PAGE_SIZE = 10;
+
+type CollectionSelectionProps = {
+    scopeId: string;
+    setFieldValue: ReturnType<typeof useFormik<{ scopeId: string }>>['setFieldValue'];
+    allowCreate: boolean;
+};
+
+function CollectionSelection({
+    scopeId,
+    setFieldValue,
+    allowCreate,
+}: CollectionSelectionProps): ReactElement {
+    const { isOpen, onToggle } = useSelectToggle();
+    const { configError, setConfigError, onSubmit } = useCollectionFormSubmission({
+        type: 'create',
+    });
+    const [search, setSearch] = useState('');
+
+    const [isCollectionModalOpen, setIsCollectionModalOpen] = useState(false);
+
+    const requestFn = useCallback(
+        (page: number) => {
+            return listCollections(
+                { 'Collection Name': search },
+                { field: 'Collection Name', reversed: false },
+                page,
+                COLLECTION_PAGE_SIZE
+            ).request;
+        },
+        [search]
+    );
+
+    const { data, isEndOfResults, isFetchingNextPage, fetchNextPage } = usePaginatedQuery(
+        requestFn,
+        COLLECTION_PAGE_SIZE
+    );
+
+    // Combines the server-side fetched pages of collections data with the local cache
```

Comment:
This section is required due to the following:
1. The collection dropdown is lazy loaded, and we likely won't have all collections client side when a new collection is created.
2. When a new collection is created, we want to have it inserted into the dropdown so that it is visible to the user.
3. After creating a collection, if the user loads more items via the "view more" button, it is likely that the duplicate collection will be returned. Since the pagination filtering happens inside the hook, we need to reconcile the difference here again.